src/modules/task-runner.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const _ = require('lodash');
+const logger = require('winston');
+
+let spawnSync = require('child_process').spawnSync;
+
+class TaskRunner {
+  constructor(task, vars) {
+    this.task = task;
+    this.vars = vars;
+  }
+
+  execute() {
+    _.forEach(this.task, command => this.runCommand(command));
+  }
+
+  runCommand(command) {
+    if(!command.cmd) throw new Error('Command not defined')
+    const parsedCommand = this.expandTokens(command.cmd).split(" ");
+    const parsedEnv = this.resolveKeyValuePairs(command.environment);
+    const spawnOptions = this.buildSpawnOptions(parsedEnv)
+
+    logger.info(`Executing command : ${parsedCommand.join(" ")}`);
+
+    const result = spawnSync(parsedCommand[0], _.tail(parsedCommand), spawnOptions);
+
+    if (command.register) {
+      this.vars[command.register] = result.stdout;
+    }
+
+    return (result.status !== 0 && !command.ignore_errors) ? false : true;
+  }
+
+  expandTokens(command) {
+    const template = _.template(command);
+
+    return template(this.vars);
+  }
+
+  resolveKeyValuePairs(array) {
+    const templated = this.vars ? _.map(array, a => this.expandTokens(a, this.vars)) : array;
```

Comment:
Could extract `_.map(array, a => this.expandTokens(a, this.vars))` to a variable to make it easier to understand.